src/main/webapp/components/workspace/filters.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react'
+import Button from '@material-ui/core/Button'
+import Divider from '@material-ui/core/Divider'
+import Section from './section'
+import Typography from '@material-ui/core/Typography'
+import { defaultFilter } from '../query-builder/filter/filter-utils'
+import Box from '@material-ui/core/Box'
+import Filter from '../query-builder/filter/individual-filter'
+
+const anyTextFilter = { ...defaultFilter, value: '*' }
+
+const FilterList = ({ filters, onChange }) =>
+  filters.map((filter, i) => (
+    <Box key={i} style={{ padding: '0px 16px' }}>
+      <Filter
+        filter={filter}
+        onChange={newFilter => {
+          // Have to copy / clone array to modify
+          const newFilters = filters.slice()
+          newFilters[i] = newFilter
+          onChange(newFilters)
+        }}
+        onRemove={() => {
+          onChange(filters.filter((filter, index) => index !== i))
+        }}
+      />
+    </Box>
+  ))
+
+const Filters = ({ onApply }) => {
+  const [filters, setFilters] = useState([])
+  const [enable, setEnable] = useState(false)
+
+  const shouldAddFilters = () => {
+    if (enable) {
+      onApply(filters)
+    }
+  }
+
+  return (
+    <Section
+      title="Filters"
+      enable={enable}
+      onChange={() => setEnable(!enable)}
+    >
+      <Box display="flex" alignItems="center" justifyContent="center">
+        <Box padding="16px">
+          <Button
+            color="primary"
+            variant="contained"
+            onClick={() => setFilters([...filters, anyTextFilter])}
+          >
+            <Typography>Add Filter</Typography>
+          </Button>
+        </Box>
+
+        <Box padding="16px">
+          <Button
+            color="primary"
+            variant="contained"
+            onClick={() => shouldAddFilters()}
```

Comment:
I'd argue that the enabled state should be set to true whenever the apply button is pressed. You can also add in a useEffect whenever enabled is toggled that reruns the query with the correct filters. What do you think?

Reply:
I'm not sure it's necessary, as you won't be able to see the Apply Filters button unless enabled is set to true. If we want to make sure that filters are enabled, i can add it.

Reply:
hmm, it doesn't look like the Apply Filters button is conditionally rendered. Where does that ^ happen?

Reply:
It's part of the children of Section

Reply:
Pretty cool, didn't know about the Collapse component :+1: Shouldn't it be safe to call the onApply without checking for the enabled state then?